### ISSUE TITLE
LoRA training takes up too much VRAM?

### ISSUE THREAD

**Issue by gameveloster** [oobabooga/text-generation-webui#2772]
I am starting to learn how to train a LoRA so please correct me if I am making a mistake.

When I load the base model `TheBloke/gpt4-alpaca-lora-30B-GPTQ-4bit-128g` with `GPTQ-for-LLaMa` loader, my VRAM usages for both 3090 cards are 9.4 GB and 10.4 GB. Ooba was started with `--monkey-patch`. 

On starting the training with `micro_batch_size=2`, they jump up to 16.3 and 19.0 GB. With `micro_batch_size=3`, VRAM usages are 19.3 GB and 22.6 GB. They OOM with `micro_batch_size=4`.

1. Are these VRAM numbers normal for what I am doing? I was hoping to be able to use just a single 24GB GPU to do the LoRA training. 
2. Is there a better way of doing the LoRA training with Ooba than the steps I have taken above?
3. Should a quantized base model be used, like the GPTQ one?
4. Should we be using a LLaMA/Alpaca base model instead?

**Comment by Ph0rk0z** [comment]
The native repo is a bit faster for this if you can set it up. Also it can use gradient checkpointing, just beware it disables dropout and you can over train. You are using a model with 128g for some reason too and that will eat more vram.

Is high vram usage normal in training; hella.

**Comment by gameveloster** [comment]
@Ph0rk0z Got it! Which repo is the native repo?
